masu/processor/celery/worker.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Celery worker module."""
+
+import logging
+
+from celery import Celery
+from flask import Flask
+
+
+def make_celery(app):
+    """Create Celery app object using the Flask app's settings."""
+    celery = Celery(
+        app.import_name,
+        backend=app.config.get('CELERY_RESULT_BACKEND'),
+        broker=app.config.get('CELERY_BROKER_URL')
+    )
+    celery.conf.update(app.config)
+
+    # pylint: disable=too-few-public-methods
+    class ContextTask(celery.Task):
+        """Celery Task Context."""
+
+        def __call__(self, *args, **kwargs):
+            """Call task with context."""
+            with app.app_context():
+                return self.run(*args, **kwargs)
+
+    celery.Task = ContextTask
+    return celery
+
+
+LOG = logging.getLogger(__name__)
+FLASK = Flask(__name__, instance_relative_config=True)
```

Comment:
You would want to create an instance of our app here, which is made using the create_app app factory, I believe? I was thinking we would create the celery app inside our app factory so that it would always be available in our app.

Reply:
I was having trouble seeing how to accomplish that in a satisfying way.

Also, `worker` is what is called by the openshift template to spawn the workers, which is a secondary entry point from the Flask app itself.